clang/lib/Driver/ToolChain.cpp

```diff
@@ -766,9 +766,19 @@ ToolChain::getTargetSubDirPath(StringRef BaseDir) const {
     return {};
   };
 
-  if (auto Path = getPathForTriple(getTriple()))
+  llvm::Triple Triple = getTriple();
+
+  // Try triple as is.
+  if (auto Path = getPathForTriple(Triple))
     return *Path;
 
+  // Match transformations in CompilerRTUtils.cmake:get_compiler_rt_target.
```

Comment:
https://reviews.llvm.org/D133406

Is this transformation only needed by Solaris? It seems that other OSes are happy with always using x86_64 and not bothering with amd64.

Can you make this Solaris specific?

Reply:
It's not, some BSDs prefer the `amd64` form, too.  Besides, `get_compiler_rt_target` does it unconditionally, so `clang` should match.

Reply:
I suspect that some BSDs used `amd64` but now switched to `x86_64`, e.g.
[x86_64-unknown-openbsd](https://doc.rust-lang.org/nightly/rustc/platform-support/openbsd.html)

It's fine that Solaris is different, but I think other OSes don't think this change.

Perhaps get_compiler_rt_target should be changed to affect only Solaris as well, but that change should not be merged to release/19.x to prevent potential disruption.

Reply:
How would this change cause harm?  If a target uses e.g. an `x86_64-*` or `sparcv9-*` triple, it matches `compiler-rt` out of the box and this change doesn't make a difference.  If on the other hand a target uses the equivalent `amd64-*` or `sparc64-*` forms, they **need** to be transformed to the form that `compiler-rt` uses unconditionally, otherwise `clang` won't find the runtime libs.  This is what this patch does: bring `clang` and `compiler-rt` a little bit more in sync, nothing more, nothing less.  Where do you see any disruption in that?  Fixing tons of link failures in the testsuite isn't my definition of disruption, actually.

Reply:
I don't see this as disruption, but I see `amd64-*` `sparcv9` as workarounds. For workarounds, we generally keep their scope as narrow as possible. We don't encourage current or future use cases.

I am concerned if supporting `amd64-*` opens the door for incorrect `amd64-linux`  usage.

Reply:
I don't agree with the third bullet point.
As is, people are fine with not using the `amd64` triples on all non-Solaris OSes.
We strive not to add workarounds that could lure users to do the wrong thing.

Note: We perform basic triple canonicalization (`Triple::normalize`), as you can see from "unknown":

```
% clang --target=x86_64-linux-gnu -dumpmachine
x86_64-unknown-linux-gnu
```

That said, we try to keep the code simple. We cannot really do every triple normalization GCC performs.
That's a lot of complexity, workarounds for old systems.
Instead, we should assess every extra rule. It seems that Solaris, a supported OS, still uses `amd64`, so we accept it (with Solaris specific condition so that we know the difference from the generic behavior; and we can drop the behavior once it becomes unneeded)
but it's not sufficient justification to extend this to all OSes.

I probably should only accepted https://reviews.llvm.org/D133406 when it was made Solaris specific.

Reply:
You still don't give any indication why one form of a triple is right and another one wrong, especially when it works in 95+% of the cases. Besides, there's no way a user can even determine what's right and and what's wrong other than spurious link failures.

We're getting absolutely nowhere with this: you keep making assertions without justification.  This is a total waste of my time, the whole issue having dragged on for about two years in various forms, amounting to nothing.